joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -1,16 +1,96 @@
 package io.joern.csharpsrc2cpg.astcreation
 
-import io.joern.csharpsrc2cpg.parser.DotNetJsonAst.LiteralExpr
+import io.joern.csharpsrc2cpg.parser.DotNetJsonAst.*
 import io.joern.csharpsrc2cpg.parser.{DotNetNodeInfo, ParserKeys}
 import io.joern.x2cpg.{Ast, ValidationMode}
+import io.shiftleft.codepropertygraph.generated.nodes.NewCall
+import io.shiftleft.codepropertygraph.generated.{DispatchTypes, Operators}
 
 trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
 
+  def astForExpression(expr: DotNetNodeInfo): Seq[Ast] = {
+    // TODO: Handle identifiers in operators
+    val expressionNode = createDotNetNodeInfo(expr.json(ParserKeys.Expression))
+    expressionNode.node match
+      case _: UnaryExpr  => astForUnaryExpression(expressionNode)
+      case _: BinaryExpr => astForBinaryExpression(expressionNode)
+      case _             => notHandledYet(expressionNode)
+  }
+
+  protected def astForLiteralExpression(_literalNode: DotNetNodeInfo): Ast = {
+    Ast(literalNode(_literalNode, code(_literalNode), nodeTypeFullName(_literalNode)))
+  }
+  private def astForUnaryExpression(unaryExpr: DotNetNodeInfo): Seq[Ast] = {
+    val operatorToken = unaryExpr.json(ParserKeys.OperatorToken)(ParserKeys.Value).str
+    val operatorName = operatorToken match
+      case "+" => Operators.plus
+      case "-" => Operators.minus
+      case "++" =>
+        if (unaryExpr.node.getClass == PostIncrementExpression.getClass) Operators.postIncrement
+        else Operators.preIncrement
+      case "--" =>
+        if (unaryExpr.node.getClass == PostDecrementExpression.getClass) Operators.postDecrement
+        else Operators.preDecrement
+      case "~" => Operators.not
+      case "!" => Operators.logicalNot
+      case "&" => Operators.addressOf
+
+    val argsAst = astForNode(unaryExpr.json(ParserKeys.Operand))
+    Seq(
+      callAst(createCallNodeForOperator(unaryExpr, operatorName, typeFullName = Some("")), argsAst)
+    ) // TODO: typeFullName
```

Comment:
Try to work on adding support for TypeFullName from the beginning which will help type information shape up well. Otherwise, you might end up doing good refactoring of the code.

Reply:
At least send it to `AstCreatorHelper.nodeTypeFullName` in the meantime. Could always modify that method to return a `None` or `Some("ANY")` if the `ParserKeys.Type` property is not found.